frontend/src/js/entity-history/LoadHistoryDropzone.tsx

```diff
@@ -46,22 +48,26 @@ export const LoadHistoryDropzone = ({
     label: string;
     data: string[][];
   }) => {
-    const loadedEntityIds = [];
+    const loadedEntityIds: EntityId[] = [];
     const loadedEntityStatus: EntityIdsStatus = {};
     const loadedEntityStatusOptionsRaw: string[] = [];
 
     for (const row of data) {
-      if (row.length !== 2) {
+      if (row.length < 2) {
         continue;
       }
 
-      loadedEntityIds.push(row[0]);
-      if (row[1]) {
-        loadedEntityStatus[row[0]] = row[1].split(",").map((s) => {
-          const opt = s.trim();
-          loadedEntityStatusOptionsRaw.push(s);
-          return { label: opt, value: opt };
-        });
+      loadedEntityIds.push({ kind: row[0], id: row[1] });
+
+      if (row.length > 2) {
+        loadedEntityStatus[row[1]] = row
+          .slice(2)
+          .filter((str) => str.length > 0)
+          .map((s) => {
+            const opt = s.trim();
+            loadedEntityStatusOptionsRaw.push(opt);
+            return { label: opt, value: opt };
+          });
```

Comment:
This section would benefit from an explanation of what `data` is and why you're preferably looking for rows of exactly length 2.

Reply:
Good point, I grouped the save and load logic + added a comment